akka-docs/rst/java/persistence.rst

```diff
@@ -427,22 +427,19 @@ this will send the message to the destination actor. When recovering, messages w
 Once recovery has completed, if there are outstanding messages that have not been confirmed (during the message replay), 
 the persistent actor will resend these before sending any other messages.
 
-Deliver also requires a function to pass the ``deliveryId`` into the message. A ``deliveryId`` is required to acknowledge 
-receipt of a message, and is also used in playback, when the actor is recovering so that messages received can be correctly acknowledged. 
-A function can be created to map your own ``messageId`` to ``deliveryId``, which may come from your own domain model. 
-This function must keep track of which ``messageId`` have been acknowledged.
-Alternatively, the Persistence module provides a default sequence number implementation which can also be used as the ``deliveryId`` 
-for messages. The default sequence increases monotonically, without gaps.
+Deliver requires a ``deliveryIdToMessage`` function to pass the ``deliveryId`` into the message so that correlation 
+between ``deliver`` and ``confirmDelivery`` is possible. The ``deliveryId`` must do the round trip. Upon receipt 
+of the message destination actor will send the same``deliveryId`` wrapped in a confirmation message back to the sender. 
+The sender will then use it to call ``confirmDelivery`` method to complete delivery routine.
 
 .. includecode:: code/docs/persistence/PersistenceDocTest.java#at-least-once-example
 
-Correlation between ``deliver`` and ``confirmDelivery`` is performed with the ``deliveryId`` that is provided
-as parameter to the ``deliveryIdToMessage`` function. The ``deliveryId`` is typically passed in the message to the
-destination, which replies with a message containing the same ``deliveryId``.
-
-The ``deliveryId`` is a strictly monotonically increasing sequence number without gaps. The same sequence is
-used for all destinations of the actor, i.e. when sending to multiple destinations the destinations will see
-gaps in the sequence if no translation is performed.
+The ``deliveryId`` that is passed into the ``deliveryIdToMessage`` function is an internal id generated by the persistence module. 
+It is a strictly monotonically increasing sequence number without gaps. The same sequence is used for all destinations of the actor, 
+i.e. when sending to multiple destinations the destinations will see gaps in the sequence if no translation is performed. 
+Translation between internal ``deliveryId`` and actual ``deliveryId`` (being for example a ``messageId`` comming from domain model) 
```

Comment:
I find this part still a bit confusing. The "**actual** deliveryId" is a bit out of context. 

How about starting out with:
```
It is possible to use custom delivery ids, however you must then retain a mapping between the internal deliveryId (passed into the deliveryIdToMessage function) to your custom delivery id. You can do this by storing such mapping in a `Map(customDeliveryId -> deliveryId)` from which you can retrive the actual delivery Id to be passed into confirmDelivery once the receiver of your message has replied with your custom delivery Id.
```

Hope this sounds better? Feel free to improve upon this, but let's explicitly say things like "custom" and explain how the mapping would be used (the process, step by step).

// Sorry that I've spread out reviewing this commit in it's entirety so long. These should be final comments here though.